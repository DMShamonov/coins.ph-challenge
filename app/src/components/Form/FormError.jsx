// @flow

import * as React from 'react';
import classnames from 'classnames';

import type { FormErrorPropsType } from './types.js.flow';

export default function FormError({
  className,
  children,
}: FormErrorPropsType): React.Node {
  return (
    <p className={classnames('form__error', className)}>{children}</p>
  );
}
