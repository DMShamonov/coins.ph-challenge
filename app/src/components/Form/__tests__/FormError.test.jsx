import React from 'react';
import { shallow } from 'enzyme';

import FormError from '../FormError';

describe('`FormError` component should render correctly', () => {
  test('with default props', () => {
    expect(shallow(<FormError>__CONTENT__</FormError>)).toMatchSnapshot();
  });

  test('with `className` prop', () => {
    expect(shallow(<FormError className="__CLASSNAME__">__CONTENT__</FormError>)).toMatchSnapshot();
  });
});
