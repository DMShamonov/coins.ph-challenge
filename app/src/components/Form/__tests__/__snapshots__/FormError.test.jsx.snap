// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`FormError\` component should render correctly with \`className\` prop 1`] = `
<p
  className="form__error __CLASSNAME__"
>
  __CONTENT__
</p>
`;

exports[`\`FormError\` component should render correctly with default props 1`] = `
<p
  className="form__error"
>
  __CONTENT__
</p>
`;
